app/bookmarks/tags/[tagSlug]/page.tsx

```
@@ -23,9 +23,11 @@ export async function generateStaticParams() {
   const tags = bookmarks.flatMap(b =>
     (Array.isArray(b.tags) ? b.tags : []).map(t => (typeof t === 'string' ? t : t.name))
   );
-  const uniqueSlugs = Array.from(new Set(tags)).map(tag =>
-    tag.toLowerCase().replace(/\s+/g, '-')
-  );
+  const uniqueSlugs = Array.from(new Set(tags)).map(tag => {
+    // Strip Unicode control characters
+    const cleanTag = tag.replace(/[\u0000-\u001F\u007F-\u009F\u200B-\u200F\u2028-\u202F\u2066-\u206F]/g, '');
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)